src/crypto.js

```diff
@@ -85,18 +99,23 @@ export default function makeCryptoWith(opts) {
       _generateKey(encryptionKey, salt)
     ])
     .then(results => {
+      _validateAADType(aad);
```

Comment:
Can we move this to the very beginning of the encrypt method? There's no reason for us to perform this work if the AAD is invalid.

Reply:
yeah, I was trying to think what happens if the error is thrown outside the promise but the recovery from the error is to change code and to call the function appropriately or pass the correct options. So might be better to throw an error on the line that calls encrypt instead of waiting till later.